### PR TITLE
Always publish a build scan in CI

### PR DESCRIPTION
**Pull request by alpar-t**
This PR changes the build scan configuration to always publish a  build
scan when running in our CI and configures the server too, to have everything in one place. 

We also now pass in key env vars to the vagrant VM to trigger the publication of build scans. 

The old properties to accept build scan ToS on the public server are
thus no longer relevant and will be cleaned up from the Jenkins config
once this is merged.

### CODE REVIEW

File: gradle/build-scan.gradle [hunk 1]
```diff
@@ -3,38 +3,28 @@ import org.elasticsearch.gradle.OS
...
 
     // Jenkins-specific build scan metadata
     if (jenkinsUrl) {
         tag 'CI'
-        tag System.getenv('JOB_NAME')
-        link 'Jenkins Build', System.getenv('BUILD_URL')
-        link 'Additional Logs',
-                "https://console.cloud.google.com/storage/elasticsearch-ci-artifacts/jobs/${jobName}/build/${buildNumber}.tar.bz2"
-        System.getenv('NODE_LABELS').split(' ').each {
+        tag jobName
```
Comment (mark-vieira) [inline gradle/build-scan.gradle]: I wonder if we should add the build number as a custom value as well, that way we could search for all scans for a given job? Or even the URL itself as a custom value would serve the same purpose, as right now you can search on links.

**Review by mark-vieira — APPROVED** [review]
LGTM 👍

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Pull request merged**